src/player/Player.php

```diff
@@ -428,19 +432,19 @@ public function getUniqueId() : UuidInterface{
 	/**
 	 * TODO: not sure this should be nullable
 	 */
-	public function getFirstPlayed() : ?int{
+	public function getFirstPlayed() : ?DateTimeImmutable{
 		return $this->firstPlayed;
 	}
 
 	/**
 	 * TODO: not sure this should be nullable
 	 */
-	public function getLastPlayed() : ?int{
+	public function getLastPlayed() : ?DateTimeImmutable{
 		return $this->lastPlayed;
 	}
 
 	public function hasPlayedBefore() : bool{
-		return $this->lastPlayed - $this->firstPlayed > 1; // microtime(true) - microtime(true) may have less than one millisecond difference
+		return ((int) $this->firstPlayed->diff($this->lastPlayed)->format('%f')) > 1;
```

Comment:
This behaviour differs from the original. The previous code checked if the difference was at least 1 second.
This is now checking if the difference is at least 1 microsecond, which is much more prone to failure.